src/config_server.c

```diff
@@ -0,0 +1,2396 @@
+    /* no ssh-x509-certs */
+    const char *ietf_ssh_common[] = {"transport-params", "public-key-generation", NULL};
+    /* all features */
+    const char *iana_ssh_encryption_algs[] = {NULL};
+    /* all features */
+    const char *iana_ssh_key_exchange_algs[] = {NULL};
+    /* all features */
+    const char *iana_ssh_mac_algs[] = {NULL};
+    /* all features */
+    const char *iana_ssh_public_key_algs[] = {NULL};
+    /* all features */
+    const char *ietf_keystore[] = {"central-keystore-supported", "local-definitions-supported", "asymmetric-keys", "symmetric-keys", NULL};
+    /* no ssh-server-keepalives and local-user-auth-hostbased */
+    const char *ietf_ssh_server[] = {"local-users-supported", "local-user-auth-publickey", "local-user-auth-password", "local-user-auth-none", NULL};
+    /* all features */
+    const char *ietf_truststore[] = {"central-truststore-supported", "local-definitions-supported", "certificates", "public-keys", NULL};
+    /* all features */
+    const char *ietf_tls_server[] = {"tls-server-keepalives", "server-ident-x509-cert", "server-ident-raw-public-key", "server-ident-tls12-psk",
+                                     "server-ident-tls13-epsk", "client-auth-supported", "client-auth-x509-cert", "client-auth-raw-public-key",
+                                     "client-auth-tls12-psk", "client-auth-tls13-epsk", NULL};
+    /* all features */
+    const char *libnetconf2_netconf_server[] = {NULL};
+
+    const char *module_names[] = {"ietf-netconf", "ietf-netconf-monitoring", "ietf-netconf-server", "ietf-x509-cert-to-name", "ietf-crypto-types",
+                                  "ietf-tcp-common", "ietf-ssh-common", "iana-ssh-encryption-algs",
+                                  "iana-ssh-key-exchange-algs", "iana-ssh-mac-algs", "iana-ssh-public-key-algs",
+                                  "ietf-keystore", "ietf-ssh-server", "ietf-truststore",
+                                  "ietf-tls-server", "libnetconf2-netconf-server", NULL};
+
+    const char **module_features[] = {ietf_netconf, ietf_netconf_monitoring, ietf_nectonf_server, ietf_x509_cert_to_name, ietf_crypto_types,
+                                      ietf_tcp_common, ietf_ssh_common, iana_ssh_encryption_algs,
+                                      iana_ssh_key_exchange_algs, iana_ssh_mac_algs, iana_ssh_public_key_algs,
+                                      ietf_keystore, ietf_ssh_server, ietf_truststore,
+                                      ietf_tls_server, libnetconf2_netconf_server, NULL};
+
+    for (i = 0; module_names[i] != NULL; i++) {
+        if (!ly_ctx_load_module(*ctx, module_names[i], NULL, module_features[i])) {
+            ERR(NULL, "Loading module \"%s\" failed.\n", module_names[i]);
+            goto error;
+        }
+    }
+
+    return 0;
+
+error:
+    if (new_ctx) {
+        ly_ctx_destroy(*ctx);
+        *ctx = NULL;
+    }
+    return 1;
+}
+
+API int
+nc_config_setup_server(const struct lyd_node *data)
```

Comment:
A variant `nc_config_setup_server_path` that would parse a file and then pass the data to this function seems useful.